lib/qrolo.dart

```diff
@@ -27,526 +28,277 @@ import 'package:qrolo/src/html/media/utilities/is_media_device_camera_available.
-        // We do not receive the extra error info here
-        // Need to use custom JS interop to expose more error info.
-        // "NotFoundError: Requested device not found"
-        // Indicates error on getUserMedia()
-        const String message = 'No camera access found: '
-            'Please check camera device/permission';
-
-        _updateErrorMessage(message);
-
-        return null;
-      }
-
-      return mediaStream;
-    } on html.DomException catch (domException, stackTrace) {
-      // Code actually breaks out exception rather than returning null
-
-      _updateErrorMessage(
-        'Unable to access camera stream \n'
-        'Please check camera devices/permissions \n'
-        'DOM Exception ${domException.toString()} ${stackTrace.toString()}',
-      );
-
-      return null;
-    } on Exception catch (e, stackTrace) {
-      _updateErrorMessage(
-        'Unable to access camera stream getUserMedia(): '
-        'Exception: ${e.toString()} ${stackTrace.toString()}',
-      );
-
-      return null;
-      // ignore: avoid_catches_without_on_clauses
-    } catch (e, stackTrace) {
-      // This should not occur
-      _updateErrorMessage(
-        'Unable to access camera stream errors: '
-        '${e.toString()} ${stackTrace.toString()}',
-      );
-
-      return null;
+      var constraints = UserMediaOptions(
+          // audio: false,
+          video: VideoOptions(
+        facingMode: (front ? "user" : "environment"),
+      ));
+      // dart style, not working properly:
+      // var stream =
+      //     await html.window.navigator.mediaDevices.getUserMedia(constraints);
+      // straight JS:
+      final stream =
+          await promiseToFuture<html.MediaStream>(getUserMedia(constraints));
+      _localStream = stream;
+      video?.srcObject = _localStream;
+      video?.setAttribute("playsinline",
+          'true'); // required to tell iOS safari we don't want fullscreen
+      final dynamic playTest = await video?.play();
+    } catch (e) {
+      print("error on getUserMedia: ${e.toString()}");
+      cancel();
+      setState(() {
+        _errorMsg = e.toString();
+      });
+      return;
     }
-  }
-
-  ///
-  /// Show the captured stream to the user on the page video element src
-  /// Test using https://media-play-promise.glitch.me/
-  /// https://webrtc.github.io/samples/src/video/chrome.mp4
-  /// videoElementToUpdateDirectly.setAttribute(
-  ///   'src',
-  ///   'https://webrtc.github.io/samples/src/video/chrome.mp4',
-  /// );
-  ///
-  /// Error: NotAllowedError: play() failed because the user didn't interact with the document first.
-  /// https://goo.gl/xX8pDD
-  Future<dynamic?> startPlayingStream(
-    html.MediaStream videoStream,
-    html.VideoElement videoElementToUpdateDirectlyz,
-  ) async {
     if (!mounted) return;
 
-    // Mutate
-    // Present the media stream on the HTMLVideoElement
-
-    this.videoElement.srcObject = videoStream;
```

Comment:
Undefined videoElement

the div was not created in the web page? (in refactored code)